lib/y2lan_restart_common.pm

```diff
@@ -149,14 +149,24 @@ sub check_network_status {
     }
     assert_script_run 'cat journal.log';                              # print journal.log
     my $is_empty = script_run('[ -s journal.log ]');
+    # empty journal.log indicating that service was not restarted or reloaded
     if ($is_empty) {
-        assert_equals($expected_status, 'no_restart', "Network restart didn't occur, even though it was expected.");
+        assert_equals($expected_status, 'no_restart', "Network service should have been restarted/reloaded. journalctl cannot find events");
     }
+    # Based on the query's filter it is assumed that the service is restarted otherwise it is reloaded.
     else {
-        assert_equals($expected_status, 'restart', "Network restart occurred, even though it wasn't expected.");
+        my $journal_findings = script_output('cat journal.log');
+        my $query_pattern    = "shutting down|ifdown all|Stopping wicked";
+        my @matched          = grep { $_ =~ /($query_pattern)/ } $journal_findings;
+        if (@matched) {
+            assert_equals($expected_status, "restart", "Network restart occurred, even though it wasn't expected.. journalctl found \n$journal_findings\n\n");
```

Comment:
I also don't get this assert, we pass parameter to the method and compare it with some hardcoded string. This is confusing as actual check is done above and we have detected restart.

Reply:
exactly, we could get the string 'restart' or 'reload' depending on the matches before doing `else/else`, for example.

Reply:
i also thought that it was confusing but it has a logic. But this is not different from what it was before. i just extend the conditions to include and distiguish from when the method expects a restart(sle12) and a reload(sle15, tw). 
The first step is to grep the logs to see if any reload/restart occured. 
Then we have
- if not filters found -> make sure that we do not expect reload/restart
- if any of the filters is in the logs -> make sure that a restart event occured or a reload (if the enties is not one of  shutting down|ifdown all|Stopping wicked)

Reply:
previous code was binary, or restart or not restart, and I noticed that check with a hardcoded value was not good idea (my bad to have merged it) but was simple enough, the problem is that is does not scale that approach with more value, it is not really readable, so if the file is not empty and we don't detect restart we will always fail thinking that is a reload what we expect, the failure will not be clear IMO. I would be better to extract expectation before the conditions.

Reply:
> if the file is not empty and we don't detect restart we will always fail thinking that is a reload what we expect

in this case the logs have found match with _Reloaded wicked_ . So yes the reload would be expected to match from the assertion correctly. 

This is actually interesting topic. for now as it is, it looks good to me. but i am open for ideas how to improve it. What do you mean saying "extract expectation before the conditions."?

Reply:
i modified the logic so now based on the expectations it try to match context with according filters